### PR TITLE
PeriodicTimer XML documentation improvement

### PR DESCRIPTION
**Pull request by theodorzoulias**
The XML documentation of the constructor of the class [`PeriodicTimer`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.periodictimer) states that the `period` must represent a number of milliseconds larger than 0, but actually values between 0.0 and 1.0 are invalid. This PR improves the text by stating that the milliseconds must be at least 1.

Minimal demonstration of the actual behavior:

```C#
using System;
using System.Threading;

public class Program
{
    public static void Main()
    {
        PeriodicTimer timer = new(TimeSpan.FromMilliseconds(0.5));
    }
}
```

Output:

```none
Unhandled exception. System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'period')
```

[Online demo](https://dotnetfiddle.net/hxApGr).

### CODE REVIEW

**Comment by stephentoub** [comment]
It's not really wrong; the number of milliseconds when treated as an integer must be greater than 0, and the first such integer is 1. Any fractional milliseconds are cast away. That said, it's fine to clarify.

File: src/libraries/System.Private.CoreLib/src/System/Threading/PeriodicTimer.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public sealed class PeriodicTimer : IDisposable
 
         /// <summary>Initializes the timer.</summary>
         /// <param name="period">The time interval between invocations of callback..</param>
-        /// <exception cref="ArgumentOutOfRangeException"><paramref name="period"/> must be represent a number of milliseconds larger than 0 and smaller than <see cref="uint.MaxValue"/>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="period"/> must represent a number of milliseconds equal or larger than 1, and smaller than <see cref="uint.MaxValue"/>.</exception>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/PeriodicTimer.cs:26]:
```suggestion
        /// <exception cref="ArgumentOutOfRangeException"><paramref name="period"/> must represent a number of milliseconds equal to or larger than 1, and smaller than <see cref="uint.MaxValue"/>.</exception>
```

**Review by stephentoub — APPROVED** [review]

**Comment by theodorzoulias** [comment]
@stephentoub are there any thoughts about making zero (0) a valid value for the `period`? Personally I have started using the `PeriodicTimer` as a substitute of [`Task.Delay`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay) (in a non-periodic fashion), because it's more lightweight and more convenient to cancel/dispose than the `Task.Delay`+`CancellationTokenSource` combo. This tiny disparity in the range of the `delay`/`period` parameters has the potential to cause some rare and difficult to find bugs.

**Comment by stephentoub** [comment]
What is the use case for that? The difference in range exists because they're used for docent things.

**Comment by theodorzoulias** [comment]
@stephentoub as an example you could take a look at [this](https://stackoverflow.com/questions/72739162/net-6-periodictimer-with-top-of-the-minute-timing/72739676#72739676 ".NET 6 PeriodicTimer with top-of-the-minute timing") StackOverflow answer. It's an implementation of a `CronosPeriodicTimer` class that has a specialized ticking behavior, and it's based internally on a series of `PeriodicTimer` instances. Initially I had implemented it with the `Task.Delay` method, but then I experimented with a `PeriodicTimer`-based implementation and I found it to be cleaner.

**Comment by stephentoub** [comment]
I don't follow. Why does that example benefit from creating a periodic timer with a period of 0?

**Comment by theodorzoulias** [comment]
@stephentoub because after switching from the `Task.Delay` to the `PeriodicTimer`, my implementation was close to be buggy. Actually it was not buggy because of a secondary reason (the `_minDelay` safery net), but while testing with zero `_minDelay` I started getting an `ArgumentOutOfRangeException` in one every 1000 calls on average. This is something easy to miss.

Making the `period` zero-friendly would make the `PeriodicTimer` class more easily interchangeable with other timer-like components of .NET.

**Pull request merged**